test/test.js

```diff
@@ -0,0 +1,177 @@
+const cont = require('../js/actions');
+let ctrl = new cont.controller();
+
+let assert = require('assert');
+let chai = require('chai');
+let chaiHttp = require('chai-http');
+chai.use(chaiHttp);
+
+let sinon = require('sinon');
+
+let original_count = ctrl.getHeroes().length;
+
+beforeEach('before', () => {
+	ctrl =  new cont.controller();
+});
+
+describe('Array', function () {
+	describe('Heroes Json has data', function () {
+		it('Hero data json should contain entries', function () {
+			assert.notEqual(0, original_count);
+		});
+	});
+
+	describe('Controller Working', function () {
+
+		describe('Add Function', function () {
+			it('Add a hero', function () {
+				ctrl.addHero('1000', 'test unit');
+				let n_count = ctrl.getHeroes().length;
+
+				assert.equal(original_count + 1, n_count);
+			});
+			it('Add a hero - already exist', function () {
+				ctrl.addHero(1000, 'test unit');
+				let count = ctrl.getHeroes().length;
+				assert.equal(count , ctrl.getHeroes().length);
+			});
+		});
+
+		describe('Delete Function',function () {
+			it("Delete hero 'test unit'", function () {
+				let Stubconst = {};
+				Stubconst.getHeroByName = sinon.stub();
+				let output = { "id": "1000", "name": "test unit" };
+				Stubconst.getHeroByName.withArgs("test unit").returns(output);
+
+				ctrl.getHeroByName = Stubconst.getHeroByName;
+				ctrl.delHero("test unit");
+
+
+				assert.equal(ctrl.getHeroes().length,original_count);
+			});
+		});
+
+		describe('Update Function',function () {
+			it('Update a hero', function () {
+				const Stubcosnt = {};
+				Stubcosnt.getHeroById = sinon.stub();
+				let output = { "id": "2", "name": "Omer" };
+				Stubcosnt.getHeroById.withArgs(2).returns(output);
+				ctrl.getHeroById = Stubcosnt.getHeroById;
+				ctrl.updateHero(2, 'kukuflitzu');
+
+
+				assert.equal(output.name, 'kukuflitzu');
+			});
+		});
+		describe('Get Functions',function () {
+			it('Get hero by Id', function () {
+				let hero = ctrl.getHeroById(1);
+				assert.notEqual(hero, undefined);
+			});
+			it('Get hero by Name', function () {
+				let hero = ctrl.getHeroByName("Saar");
+				assert.notEqual(hero, undefined);
+			});
+		});
+
+	});
+
+	describe('Web GET requests', function () {
```

Comment:
Those tests seem like API ones. Basically dont forget splitting things to files. In anything including tests